Espg16-129 create model classes for data ryan

ryandavidsutton opened the pull request:
Model classes completed - some little changes need to be made for TODO's in the model classes but from communication with emi the current version is sufficient to be worked ontop of

ryandavidsutton committed:
ESPG16-129 Created templates from UML diagram

Used the UML diagram to create class template attributes ad methods. Needs more work I have highlighted TODOs

ryandavidsutton committed:
ESPG16-133 Updated List datatype to ArrayList

Changed all instances of List datatype to ArrayList

penguoir commented on app/src/main/java/com/example/espproject/Food.java:
```diff
@@ -1,6 +1,72 @@
+import java.util.Dictionary;
+import java.util.ArrayList;
 
+public class Food { // TODO: check access modifiers.
```
What does this TODO mean?

ryandavidsutton replied:
Access modifiers are the 'private', 'protected', 'public' - I was unsure whether they were correct for each variable so I wrote that to help the next developer check. But:

Talked to emi about it and she said that all the access modifiers are ok so I'll edit the TODO line out.

penguoir commented on app/src/main/java/com/example/espproject/Food.java:
```diff
@@ -1,6 +1,72 @@
+public class Food { // TODO: check access modifiers.
+    private int foodId;
+    private String name;
+    private ArrayList<String> dietaryInfo;
```
I don't think this type is correct. Is one dietaryInfo a string?

penguoir replied:
Maybe it's food? Allergies?

ryandavidsutton replied:
@ekc30 Any input on this?

ryandavidsutton replied:
Or @ange-02?

penguoir commented on app/src/main/java/com/example/espproject/Food.java:
```diff
@@ -1,6 +1,72 @@
+    private String name;
+    private ArrayList<String> dietaryInfo;
+    private int unit;
+    private Dictionary<String, Integer> nutritional_info;
```
should be `nutritionalInfo`, why is this a String -> Integer map?

ryandavidsutton replied:
Changed the name but again I am not sure on the actual content of this variable.
@ekc30 ?

ryandavidsutton replied:
Or @ange-02?

ange-02 replied:
Imo, nutritional info can be a json of the form 

{
    sugars:  int,
    carbs: int
...
}
Doing it as a list could also work if we standardise what each index in the list refers to, but it's a bit hacky and hard to expand.

ange-02 replied:
Also, does dietary info refer to what allergies it could trigger?

ryandavidsutton replied:
> Imo, nutritional info can be a json of the form
> 
> { sugars: int, carbs: int ... } Doing it as a list could also work if we standardise what each index in the list refers to, but it's a bit hacky and hard to expand.

Changed datatype to json.

ryandavidsutton replied:
> > Imo, nutritional info can be a json of the form
> > { sugars: int, carbs: int ... } Doing it as a list could also work if we standardise what each index in the list refers to, but it's a bit hacky and hard to expand.
> 
> Changed datatype to json.

Ok nvm this is kinda complicated for some reason

ryandavidsutton replied:
Apparantely you need to download a .jar file then add it to the classpath/ use maven (either of which I'm not sure how to do for this github file) and then do something with a file called pom.xml?

I'm lost. But here's the resources I found:
https://www.geeksforgeeks.org/working-with-json-data-in-java/
https://www.tutorialspoint.com/json_simple/json_simple_environment.htm

penguoir replied:
JSON isn't a type, it's a way of saving data. I agree with Andrew's comment, but we could use a Java Hashmap.

ryandavidsutton replied:
Finished changes

penguoir commented on app/src/main/java/com/example/espproject/Pantry.java:
```diff
@@ -1,14 +1,17 @@
+public class Pantry { // TODO: check access modifiers.
     public static final String fireStoreCollectionName = "pantries";
-    private int id;
+    private int pantryId; // Changed from id to pantryId (per UML diagram)
```
No need to stick to the UML diagram. `id` is more concise and prevents code stutter:

ryandavidsutton replied:
Sure changed it back.

penguoir commented on app/src/main/java/com/example/espproject/Pantry.java:
```diff
@@ -1,14 +1,17 @@
-    private int id;
+    private int pantryId; // Changed from id to pantryId (per UML diagram)
     private String name;
+    // New added attributes
```
What does this mean?

ryandavidsutton replied:
Some of the files had previous code in them - I marked all the things I added as in the UML diagram under comments like those.

penguoir commented on app/src/main/java/com/example/espproject/Pantry.java:
```diff
@@ -1,14 +1,17 @@
+    private int pantryId; // Changed from id to pantryId (per UML diagram)
     private String name;
+    // New added attributes
+    private ArrayList<User> userArrayList;
```
What does `userArrayList` mean?

ryandavidsutton replied:
I changed it to userList <- I refactored all instances of List to ArrayList when updating data types (I am not sure what this variable is used for though).

penguoir commented on app/src/main/java/com/example/espproject/Pantry.java:
```diff
@@ -41,4 +44,24 @@ public String getName() {
         this.name = name;
     }
+
+    // New Attribute Getters and Setters
```
What are "New attribute"s ?

ryandavidsutton replied:
As mentioned on the ticket above. 'New Attributes' are ones that were not already existing in the java files when I got access to them. (Some variables already were implemented with getters and setters before I edited the files).

penguoir commented on app/src/main/java/com/example/espproject/Recipe.java:
```diff
@@ -1,5 +1,94 @@
+    private String imageURL;
+    private int duration; // In minutes
+    private ArrayList<String> dietaryInfo;
+    private ArrayList<String> ingredients;
```
Array of Strings?

ryandavidsutton replied:
@ekc30 @ange-02 would be the best people to ask for this - I'm not sure.

ange-02 replied:
Imo, they both can be Arrays of strings, because these are really only used to populate the UI. 
When retrieving a recipe's ingredients from the database, we can store them in a Java array and then place them in a RecyclerView. We don't need anything fancy like a dictionary as that would complicate things unnecessarily. Also, I presume the strings would be stored as "100g chicken", right?
As for dietary info, same points as above, not 100% clear what it is.

penguoir commented on app/src/main/java/com/example/espproject/Recipe.java:
```diff
@@ -1,5 +1,94 @@
+    private String name;
+    private String imageURL;
+    private int duration; // In minutes
+    private ArrayList<String> dietaryInfo;
```
Array of strings?

ryandavidsutton replied:
Again @ekc30 @ange-02, do you have any input?

penguoir commented:
I have more comments, but I think you get the gist: The UML diagram was just a building block, from here we should choose the best types, especially if this is the base of all other business logic code.

ryandavidsutton committed:
Updated Food.java to have HashMap instead of Dict

Updated Food.java to have HashMap instead of Dictionary type for nutritionalInfo

ryandavidsutton commented:
@penguoir Should be able to merge this now no?
